libimage/runtime.go

```diff
@@ -81,8 +81,13 @@ func RuntimeFromStore(store storage.Store, options *RuntimeOptions) (*Runtime, e
 	var systemContext types.SystemContext
 	if options.SystemContext != nil {
 		systemContext = *options.SystemContext
+		if systemContext.BigFilesTemporaryDir == "" {
+			systemContext.BigFilesTemporaryDir = tmpdir()
+		}
 	} else {
-		systemContext = types.SystemContext{}
+		systemContext = types.SystemContext{
+			BigFilesTemporaryDir: tmpdir(),
+		}
 	}
```

Comment:
```go
	systemContext := types.SystemContext{}
	if options.SystemContext != nil {
		systemContext = *options.SystemContext
	}
	if systemContext.BigFilesTemporaryDir == "" {
		systemContext.BigFilesTemporaryDir = tmpdir()
	}
```
might be a bit easier to follow and less repetitive.

(Personally I’d prefer a more explicit than “handle correctly” — to make it clear what was broken before and how it is fixed now. Are the two changes fixing two different bugs? But that’s depending on standards of this repo.)